File: src/test/kotlin/tracking/website/FurAffinityProfileStrategyTest.kt
package tracking.website

import kotlin.test.Test
import kotlin.test.assertEquals

class FurAffinityProfileStrategyTest {
    private val subject = FurAffinityProfileStrategy

    @Test
    fun `Simplified, working scenario`() {
        val input = "<html><head><body><div id=\"page-userpage\"><div class=\"userpage-profile\">Expected description</div></div></body></head></html>"
        val result = subject.filterContents(input)

        assertEquals("Expected description", result)
    }

    @Test
    fun `Failed matching`() {
        val input = "<html><head><body><div id=\"page-userpage\"><div class=\"wrong-class\">Description</div></div></body></head></html>"
        val result = subject.filterContents(input)

        assertEquals(input, result)
    }
}

File: symfony/tests/Utils/Web/UrlStrategy/FurAffinityProfileStrategyTest.php
<?php

declare(strict_types=1);

namespace App\Tests\Utils\Web\UrlStrategy;

use App\Utils\Web\UrlStrategy\FurAffinityProfileStrategy;
use Override;
use PHPUnit\Framework\Attributes\Small;
use PHPUnit\Framework\TestCase;

#[Small]
class FurAffinityProfileStrategyTest extends TestCase
{
    private FurAffinityProfileStrategy $subject;

    #[Override]
    protected function setUp(): void
    {
        parent::setUp();

        $this->subject = new FurAffinityProfileStrategy();
    }

    public function testSimplifiedWorkingScenario(): void
    {
        $input = '<html><head><body><div id="page-userpage"><div class="userpage-profile">Expected description</div></div></body></head></html>';
        $result = $this->subject->filterContents($input);

        self::assertSame('Expected description', $result);
    }

    public function testFailedMatching(): void
    {
        $input = '<html><head><body><div id="page-userpage"><div class="wrong-class">Description</div></div></body></head></html>';
        $result = $this->subject->filterContents($input);

        self::assertSame($input, $result);
    }
}
